src/AutoRest.CSharp/Common/Output/Models/Types/ObjectTypeProperty.cs

```diff
@@ -34,5 +36,24 @@ public ObjectTypeProperty(MemberDeclarationOptions declaration, string descripti
         /// </summary>
         public CSharpType ValueType { get; }
         public bool IsReadOnly { get; }
+
+        public bool IsSinglePropertyObject([MaybeNullWhen(false)] out ObjectTypeProperty innerProperty)
+        {
+            innerProperty = null;
+
+            if (this.ValueType.IsFrameworkType)
+                return false;
+
+            if (this.ValueType.Implementation is not ObjectType objType)
+                return false;
+
+            var properties = objType.EnumerateHierarchy().SelectMany(obj => obj.Properties);
+            bool isSingleProperty = properties.Count() == 1;
+
+            if (isSingleProperty)
+                innerProperty = properties.First();
+
+            return isSingleProperty;
+        }
     }
```

Comment:
yes, this is moved from the ModelWriter  with no change.   
Since the ObjectTypeProperty itself is internal, I guess there is no accessibility difference between using 'public' or 'internal' here? 
And all existing members in ObjectTypeProperty  is public,  so using 'public' here to align the declaration style?